nodejs/week3/warmup/index.js

```diff
@@ -0,0 +1,52 @@
+const express = require("express");
+const app = express();
+
+app.use(express.urlencoded({
+    extended: true
+}))
+
+app.get("/", (req, res) => res.send("nodejs week3 homework"));
+
+app.get("/calculator/:operator", (req, res) => {
+    const calculatorOperator = req.params.operator;
+    if (Object.keys(req.query).length > 0) {
+        const digits = Object.values(req.query).flat().map(number => Number(number));
+        let total = 0;
+        if (calculatorOperator === "add") {
+            total = digits.reduce((acc, value) => acc + value)
+        } else if (calculatorOperator === "substract") {
+            total = digits.reduce((acc, value) => acc - value);
+        } else if (calculatorOperator === "multiply") {
+            total = digits.reduce((acc, value) => acc * value)
+        } else if (calculatorOperator === "divide") {
+            total = digits.reduce((acc, value) => acc / value);
+        }
+        res.send(total.toString())
+        return;
+    } else {
+        res.send("0");
+    }
+});
+
+app.post("/calculator/:operator", (req, res) => {
+    const calculatorOperator = req.params.operator;
+    if (Object.keys(req.body).length > 0) {
+        const digits = Object.values(req.body).flat().map(number => Number(number));
+        let total = 0;
+        if (calculatorOperator === "add") {
+            total = digits.reduce((acc, value) => acc + value)
+        } else if (calculatorOperator === "substract") {
```

Comment:
```suggestion
        } else if (calculatorOperator === "subtract") {
```

Reply:
Thank you for your feedback, now my  calculator checks for numbers and respond with an error if one parameter is NaN.